### PR TITLE
Add to CORS allowed headers list

### LINKED ISSUE #529: CORS errors on Safari
# Problem

Cross Origin requests from `www` to `/users` and `/token` on `api` are rejected in Safari

## Subtasks
- [ ] Find problem
- [ ] Write failing test
- [ ] Fix problem

## References

Might be related to Safari attaching an `x-requested-with` header, as [described here](http://stackoverflow.com/questions/16824661/cors-request-not-working-in-safari)

### PR DESCRIPTION
**Pull request by rossta**
# What's in this PR?

First issue I noticed in the site was I couldn't log in; CORS appears to be broken in Safari. This PR adds two additional `access-control-request-headers`:  `origin` and `x-requested-with` which allows `OPTIONS` requests to succeed.

## References
Fixes #529

### CODE REVIEW

File: test/lib/code_corps/endpoint_test.ex [hunk 1]
```diff
@@ -0,0 +1,31 @@
+defmodule CodeCorps.EndpointTest do
+  use CodeCorps.ConnCase
```
Comment (joshsmith) [inline test/lib/code_corps/endpoint_test.ex]: This is great! I believe there's a `PageControllerTest` that's doing some CORS testing. Would you mind folding that into this test (or eliminating it, if you deem it necessary)?

File: lib/code_corps/endpoint.ex [hunk 2]
```diff
@@ -40,7 +40,7 @@ defmodule CodeCorps.Endpoint do
 
   plug Corsica, [
     origins: Application.get_env(:code_corps, :allowed_origins),
-    allow_headers: ["accept", "authorization", "content-type"],
+    allow_headers: ["accept", "authorization", "content-type", "origin", "x-requested-with"],
```
Comment (joshsmith) [inline lib/code_corps/endpoint.ex:43]: Excellent. The Safari bug had been...bugging me.

File: test/lib/code_corps/endpoint_test.ex (hunk 1 above)
Comment (rossta) [reply test/lib/code_corps/endpoint_test.ex]: Updated

**Review by joshsmith — APPROVED** [review]

**Comment by joshsmith** [comment]
@rossta I'm blocked merging until Circle is fixed. I don't know what's wrong with them right now but they appear to be just completely swallowing PR-based builds altogether.

**Comment by joshsmith** [comment]
Good work on this. 🙌

**Pull request merged**